src/plugins/bfetch/server/streaming/create_streaming_response_stream.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Stream, PassThrough } from 'stream';
+import { StreamingResponseHandler } from '../../common/types';
+
+const delimiter = '\n';
+
+export const createStreamingResponseStream = <Payload, Response>(
+  payload: Payload,
+  handler: StreamingResponseHandler<Payload, Response>
+): Stream => {
+  const stream = new PassThrough();
+  const results = handler.onRequest(payload);
+
+  results.subscribe({
+    next: (message: Response) => {
+      try {
+        const line = JSON.stringify(message);
+        stream.write(`${line}${delimiter}`);
+      } catch (error) {
+        // eslint-disable-next-line
+        console.error('Could not serialize or stream a message.')
```

Comment:
We probably should be using Kibana logging here. Also, should we do more than just log and swallow the error? Maybe we should add support for streaming an error back as part of a response? This seems likely to lead to pretty difficult to grok client behavior.

Reply:
Big +1 about sending the error back.
We could also consider retries?